### PR TITLE
feat(server): make returned books have new code

### PR DESCRIPTION
**Pull request by SirAuron**
Resolves this [ClickUp task](https://app.clickup.com/t/86bywc0qk).

### CODE REVIEW

**Comment by SirAuron** [comment]
@andrea-cerlini when you have a minute can you control this one and make sure that we hav not introduce the changes of your two commits here already? Thanks

File: packages/server/src/modules/cart/cart.resolver.ts [hunk 1]
```diff
@@ -149,7 +149,10 @@ export class CartResolver {
 
       book = bookDetails;
 
-      if (bookDetails.requests.length === 0) {
+      if (
+        bookDetails.requests.filter(({ deletedAt }) => deletedAt === null)
+          .length === 0
+      ) {
```
Comment (yusufkandemir) [inline packages/server/src/modules/cart/cart.resolver.ts:155]: image

```suggestion
      if (bookDetails.requests.length === 0) {
```

Comment (SirAuron) [reply packages/server/src/modules/cart/cart.resolver.ts:155]: Lol you are right, going to accept your suggestion yep

File: packages/server/src/modules/reservation/reservation.resolver.ts [hunk 2]
```diff
@@ -207,7 +210,10 @@ export class ReservationResolver {
 
     if (
       books.some(
-        ({ requests }) => requests.length > 0 && requests[0].saleId !== null,
+        ({ requests }) =>
+          requests.length > 0 &&
+          requests[0].saleId !== null &&
+          requests[0].sale?.refundedAt === null,
```
[the remark this answers is not in the document]
Comment (SirAuron) [reply packages/server/src/modules/reservation/reservation.resolver.ts:216]: Yep I agree, going with the second proposal👍

**Pull request merged**